lib/mqtt/mqtt.cc

```diff
@@ -575,8 +582,14 @@ SObj mqtt_connect(Timeout                    *t,
 	  reinterpret_cast<MQTTPubAckInfo_t *>(&context->variableLengthData);
 	MQTTPubAckInfo_t *outgoingPublishes =
 	  incomingPublishes + incomingPublishCount;
-	uint8_t *networkBuffer = reinterpret_cast<uint8_t *>(outgoingPublishes) +
-	                         sizeof(MQTTPubAckInfo_t) * outgoingPublishCount;
+	uint8_t *networkBuffer =
```

Comment:
Nit but for future readers it may be nice to add a comment here to tell that we need a separate allocation because the total size of a sealed object is limited and embedding this into the `CHERIoTMqttContext` object we allocated earlier would prevent us from sending/receiving large payloads.

Reply:
This could be reworked now that https://github.com/microsoft/cheriot-rtos/pull/244 has landed. 🎉